src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -425,9 +430,11 @@ ucs_status_t uct_dc_mlx5_ep_put_short(uct_ep_h tl_ep, const void *payload,
 
     UCT_CHECK_LENGTH(length, 0, iface->super.dm.seg_len, "put_short");
     UCT_DC_MLX5_CHECK_RMA_RES(iface, ep);
+    UCT_DC_MLX5_IFACE_TXQP_GET(iface, ep, txqp, txwq);
+    uct_rc_mlx5_ep_fence_put(&iface->super, txwq, &rkey, &remote_addr,
+                             ep->atomic_mr_offset, &fm_ce_se);
```

Comment:
seems excess overhead to get txqp/txwq:
- txqp is not used
- txwq is already resolved in uct_dc_mlx5_ep_short_dm() for example

maybe push uct_rc_mlx5_ep_fence_put() down to the function which resolves txwq, or as an alternative pass txqp/txwq down to these function instead of doing UCT_DC_MLX5_IFACE_TXQP_GET again inside them?

Reply:
reusing UCT_DC_MLX5_IFACE_TXQP_GET along the static flow should be priceless since optimizer will compact it anyway